log-viewer/src/main/java/com/logviewer/data2/Log.java

```diff
@@ -141,6 +148,25 @@ public Snapshot createSnapshot() {
         }
     }
 
+    private void decompressAndCopyGZipFile() throws IOException {
+
+        File tempFile = File.createTempFile("log-viewer-", file.getName(file.getNameCount() - 1) + ".tmp");
```

Comment:
Logs may be located in different directories but have the same name, log-viewer will write them into one file. Let's add a hash of the full path of the log to the temporary file name.

Reply:
By default, the `createTempFile` will generate a random number as part of the file name, e.g. `log-viewer-3225545540911384614-myapp.log.2022-05-17.0.gz.tmp`